automata/fa/dfa.py

```diff
@@ -702,26 +860,45 @@ def _find_state(
         return any(target_state_fn(state) for state in bfs_states)
 
     @staticmethod
-    def _cross_product(lhs: DFA, rhs: DFA) -> Tuple[DFAStateT, ExpandStateFn]:
-        """Builds the cross product between the two DFAs"""
+    def _cross_product(
```

Comment:
I think the fix for this is actually pretty straightforward, just iterate through all input symbols in both inner dictionaries instead of the input symbol set each time in the inner loop.

Reply:
This is already good enough, but would till be a problem if one of the DFAs is dense

Reply:
Also minor comment: There are some TODOs introduced, I think they can be removed or we create an issue for them

Reply:
I think that efficiency issue is very specific to the intersection with dense DFAs. I'll leave a Todo that this can possibly be improved by writing a helper function to case on the booleans used as inputs, and we can implement this if anyone ever comes along with this issue.

As for the todos, they were mainly for myself as I didn't write tests for a couple of edge cases with partial DFAs. I'll remove those and add tests later today at some point.

Reply:
Simplified logic and added comment per discussion: https://github.com/caleb531/automata/pull/147/commits/85d8be7b4c8d584eb2f548062b3736959495381e